galaxy/api/v2/tests/test_task_serializers.py

```diff
@@ -0,0 +1,135 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+from collections import OrderedDict
+from unittest import mock
+import uuid
+
+import pytest
+from django.utils import timezone
+from pulpcore import constants as pulp_const
+from pulpcore.app import models as pulp_models
+from rest_framework import exceptions as drf_exc
+import semantic_version
+
+from galaxy.main import models
+from galaxy.api.v2 import serializers
+
+
+class TestCollectionUploadSerializer:
+
+    def test_ok(self):
+        fp = mock.Mock()
+        fp.name = 'testnamespace-testcollection-1.2.3.tar.gz'
+        fp.size = 42
+
+        serializer = serializers.CollectionUploadSerializer(data={'file': fp})
+        serializer.is_valid(raise_exception=True)
+
+        filename = serializer.validated_data['filename']
+        assert filename.namespace == 'testnamespace'
+        assert filename.name == 'testcollection'
+        assert filename.version == semantic_version.Version('1.2.3')
+
+    def test_fail_invalid_flename(self):
+        fp = mock.Mock()
+        fp.name = 'testnamespace-testcollection.tar.gz'
+        fp.size = 42
+
+        serializer = serializers.CollectionUploadSerializer(data={'file': fp})
+
+        with pytest.raises(drf_exc.ValidationError) as excinfo:
+            serializer.is_valid(raise_exception=True)
+
+        assert excinfo.value.detail['non_field_errors'] == [
+            'Invalid filename. Expected: {namespace}-{name}-{version}.tar.gz']
+
+
+def test_collection_import_serializer():
```

Comment:
Just curious, but why the inconsistent use of class-based vs function-based tests? Most test suites I've ever worked with tend to stick with one for the sake of consistency.

Reply:
Agreed